app/models/agents/website_agent.rb

```diff
@@ -343,7 +343,7 @@ def receive(incoming_events)
               if url_template = options['url_from_event'].presence
```

Comment:
Wouldn't it be possible to write a migration which changes the options of the agent?

When `url_from_event` option is present and `data_from_event` is blank, set `url` to the value of `url_from_event` and default to `interpolated['url']` in this block.